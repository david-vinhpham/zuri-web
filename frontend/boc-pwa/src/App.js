import React from 'react';
import Loadable from 'react-loadable';
import logo from './logo.svg';
import './App.css';

const AsyncComponent = Loadable({
  loader: () => import('./components/Loading'),
  loading: () => (<div>this is loadable loading</div>),
});

function App() {
  return (
    <div className="App">
      {/*<header className="App-header">*/}
      {/*  <img src={logo} className="App-logo" alt="logo" />*/}
      {/*  <p>*/}
      {/*    Edit <code>src/App.js</code> and save to reload.*/}
      {/*  </p>*/}
      {/*  <a*/}
      {/*    className="App-link"*/}
      {/*    href="https://reactjs.org"*/}
      {/*    target="_blank"*/}
      {/*    rel="noopener noreferrer"*/}
      {/*  >*/}
      {/*    Learn React*/}
      {/*  </a>*/}
      {/*</header>*/}
      <AsyncComponent />
    </div>
  );
}

export default App;
