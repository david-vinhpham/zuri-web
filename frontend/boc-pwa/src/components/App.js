import React from 'react';
import PropTypes from 'prop-types';
import Error from 'components/Error';
import Loading from 'components/Loading';
import { Provider as ReduxProvider } from 'react-redux';

const ContextType = {
  // Enables critical path CSS rendering
  // https://github.com/BOCVN/isomorphic-style-loader
  insertCss: PropTypes.func.isRequired,
  // Universal HTTP client
  fetch: PropTypes.func.isRequired,
  pathname: PropTypes.string.isRequired,
  query: PropTypes.object,
  // Integrate Redux
  // http://redux.js.org/docs/basics/UsageWithReact.html
  ...ReduxProvider.childContextTypes,
  // ReactIntl
  locale: PropTypes.string,
  store: PropTypes.object.isRequired,
};

class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };
  static childContextTypes = ContextType;

  getChildContext() {
    return this.props.context;
  }

  render() {
    const { context, children } = this.props;
    const { store } = context;

    return (
      <ReduxProvider store={store}>
        <>
          <Error />
          <Loading />
          {React.Children.only(children)}
        </>
      </ReduxProvider>
    );
  }
}

export default App;
