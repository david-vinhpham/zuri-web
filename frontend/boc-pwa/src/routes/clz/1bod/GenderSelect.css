:root {
  --unit: 1rem;
  --item-info-color: #767676;
  --item-info-height: calc(2.5 * var(--unit));
  --item-info-border-bottom: calc(0.125 * var(--unit));
  --color-border-input: #dcdcdc;
}

.itemInfo {
  display: flex;
  height: var(--item-info-height);
  align-items: center;
  margin-top: var(--unit);
  border-bottom: var(--item-info-border-bottom) solid var(--color-border-input);
  color: var(--item-info-color);
  font-size: var(--f-text);
  font-weight: var(--font-weight-normal);
}

.genderGroup {
  display: flex;
  flex: 1;
  justify-content: flex-end;

  & label {
    margin-bottom: 0;
    margin-left: var(--unit);
  }
}

.icon {
  color: var(--c-brand);
  width: var(--b-space-2xl);
  padding-left: var(--b-space-xs);
}
