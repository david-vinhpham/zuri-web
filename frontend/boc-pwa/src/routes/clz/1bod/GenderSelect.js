import React, { Component } from 'react';
import { func, string, bool } from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import classNames from 'classnames';
import { RadioButton, RadioGroup } from 'homecredit-ui';

import s from './GenderSelect.css';

class GenderSelect extends Component {
  static propTypes = {
    onChange: func.isRequired,
    name: string.isRequired,
    value: string.isRequired,
    label: string.isRequired,
    icon: bool,
    require: bool,
    className: string,
  };

  static defaultProps = {
    icon: false,
    require: false,
    className: '',
  };

  handleChange = value => {
    this.props.onChange && this.props.onChange(this.props.name, value);
  };

  render() {
    const { value, name, label, icon, require, className } = this.props;
    const iconClassName = classNames('material-icons', { [s.icon]: icon });

    return (
      <div className={classNames(s.itemInfo, className)}>
        {icon && <i className={iconClassName}>supervisor_account</i>}
        <span>
          {label}
          {require && <sup>(*)</sup>}
        </span>
        <RadioGroup
          name={name}
          value={value}
          onChange={this.handleChange}
          className={s.genderGroup}
        >
          <RadioButton label="Nam" value="MALE" />
          <RadioButton label="Ná»¯" value="FEMALE" />
        </RadioGroup>
      </div>
    );
  }
}

export default withStyles(s)(GenderSelect);
