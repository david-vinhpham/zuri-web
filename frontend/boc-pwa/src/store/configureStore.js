import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { persistStore, persistReducer } from 'redux-persist';

import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

import storage from 'redux-persist/lib/storage';
import { name, version } from '../../package.json';
import rootReducer from '../reducers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const middleware = [thunk.withExtraArgument(helpersConfig)];

  const persistConfig = {
    key: 'BOCVN',
    storage,
    stateReconciler: autoMergeLevel2,
    blacklist: ['clz'],
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
    const composeEnhancers = composeWithDevTools({
      // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
      name: `${name}@${version}`,
    });

    // https://redux.js.org/docs/api/applyMiddleware.html
    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // https://redux.js.org/docs/api/createStore.html
  const store = createStore(persistedReducer, enhancer);

  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // Don't forget to remove `()` if you change reducers back to normal rootReducer.
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default()),
    );
  }

  const persistor = persistStore(store);

  return { store, persistor };
}
