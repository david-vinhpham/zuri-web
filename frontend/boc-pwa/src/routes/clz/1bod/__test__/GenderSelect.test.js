/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import App from 'components/App';
import GenderSelect from '../GenderSelect';

const mockStore = configureStore();
const initialState = {
  runtime: {
    availableLocales: ['en-US'],
  },
};

describe('GenderSelect', () => {
  it('renders children correctly', () => {
    const store = mockStore(initialState);
    const props = {
      onChange: jest.fn(),
      name: 'Male',
      value: 'male',
      label: '',
    };

    const wrapper = renderer
      .create(
        <App
          context={{
            insertCss: () => {},
            fetch: () => {},
            pathname: '',
            store,
          }}
        >
          <GenderSelect {...props}>
            <div className="child" />
          </GenderSelect>
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('should set values is correctly', () => {
    const props = {
      onChange: jest.fn(),
      name: '',
      value: '',
      label: '',
    };
    const wrapper = shallow(<GenderSelect {...props} />);
    wrapper.instance().handleChange();

    expect(props.onChange).toHaveBeenCalled();
  });
});
