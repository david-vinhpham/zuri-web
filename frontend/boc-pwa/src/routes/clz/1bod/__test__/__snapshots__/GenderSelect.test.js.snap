// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenderSelect renders children correctly 1`] = `
<div
  className="itemInfo"
>
  <span>
    
  </span>
  <div
    className="genderGroup"
  >
    <label
      className="Radio_field__uVamM"
    >
      <input
        checked={false}
        className="Radio_input__2Px7t"
        disabled={false}
        label="Nam"
        name=""
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        readOnly={true}
        type="radio"
        value="MALE"
      />
      <div
        className="Radio_radio__2IEpG"
        onMouseDown={[Function]}
        role="button"
        tabIndex="0"
      />
      <span
        className="Radio_text__1xEZE"
      >
        Nam
      </span>
    </label>
    <label
      className="Radio_field__uVamM"
    >
      <input
        checked={false}
        className="Radio_input__2Px7t"
        disabled={false}
        label="Nữ"
        name=""
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        readOnly={true}
        type="radio"
        value="FEMALE"
      />
      <div
        className="Radio_radio__2IEpG"
        onMouseDown={[Function]}
        role="button"
        tabIndex="0"
      />
      <span
        className="Radio_text__1xEZE"
      >
        Nữ
      </span>
    </label>
  </div>
</div>
`;
